### ISSUE TITLE
Odd behaviour of StepTo controller for small time steps

### ISSUE THREAD

**Issue by aidancrilly** [patrick-kidger/diffrax#657]
I have found some odd behaviour of the StepTo controller for small time steps, the result is also floating point precision dependent.

Below is a minimum working example. For single precision, the time jumps straight to t1 after 1 time step and the final integration result is incorrect.

``` python 3
import diffrax
import jax.numpy as jnp
import jax

t0 = 0.0
t1 = 1e-9

dt = 2e-12
dt_max = 1e-11
step_t = t0
step_ts = [step_t]
while(step_t < t1):
    step_t += dt
    step_ts.append(step_t)
    dt = min(2*dt,dt_max)

step_ts[-1] = t1
print(step_ts)

stepsize_controller = diffrax.StepTo(ts=step_ts)

def f(t,y,args):
    jax.debug.print('{t}',t=t)
    return -y/t1

sol = diffrax.diffeqsolve(terms=diffrax.ODETerm(f),
                    solver=diffrax.Euler(),
                    t0=t0,
                    t1=t1,
                    dt0=None,
                    y0=jnp.ones(1),
                    stepsize_controller=stepsize_controller)

print(sol.ts,sol.ys)
```

If I run in double using:
``` python3
from jax import config

config.update("jax_enable_x64", True)
```

Things are marginally better but the jax.debug.print still suggests that the final few times are clipped to t1.

If I swap t1 to 1.0 and change dt and dt_max accordingly, the correct behaviour is restored. I would however like to run with time steps of this order (nanoseconds) for physics simulation. Cheers!

**Comment by patrick-kidger** [comment]
Hi there! So this also came up recently in #632, which also discusses the reasoning why this happens. 

This isn't ideal for use-cases like yours, however, and I'd definitely welcome thoughts(/PRs) on how this might be improved.

**Comment by aidancrilly** [comment]
Ah apologies for missing that very similar issue (and from someone in my same field!).

Reasons make sense. For now, I'll probably go the quick route of just externally rescale the time axis to 0 to 1 and apply a correction to the ode term to compensate for this before passing to diffrax.

I don't know if it would be possible to, internally to diffeqsolve, linearly rescale the time axis to [0,1] for the integration and interpolation. This would seem easy (in principle! Seems daunting to me when I glance into the code) for ODEs but I'm not familiar with SDEs or CDEs. Does this rescaling sort these issues for all use cases?

**Comment by patrick-kidger** [comment]
> Does this rescaling sort these issues for all use cases?

Yup, it can. Ones that can be worked around, but it's easier to just use the times provided instead 😄

**Comment by aidancrilly** [comment]
Thanks, seems like the discussion is taking place on #632 so I will close this issue!
